src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -447,7 +447,7 @@ public virtual FieldType AddField(FieldType fieldType)
             }
             catch (ArgumentOutOfRangeException exp)
             {
-                throw new ArgumentException($"The GraphQL type for field '{Name ?? GetType().Name}.{name}' could not be derived implicitly from type '{typeof(TReturnType).Name}'.", exp);
+                throw new ArgumentException($"The GraphQL type for field '{Name ?? GetType().Name}.{name}' could not be derived implicitly from type '{typeof(TReturnType).Name}'. " + exp.Message, exp);
```

Comment:
Are you sure it's a good practice to append inner exception message here? Generally I try to avoid this.

Reply:
In this case, the inner exception messages are always generated by our own code and explain the cause of the problem.  See tests which demonstrate how the exception messages are now more detailed.

Reply:
Notice also, we are not throwing an `ExecutionError` that may be passed to clients.  This is just rethrowing the exception with additional detail specifying which field/argument and type caused the problem.

Reply:
Basically, these errors will always be caught during development time (or graph initialization would fail), and the more detailed the exception message, the better.

Reply:
I see that. I mean that exception's information is "duplicated" across parent exception and inner exception. Not a big deal for me.

Reply:
True, but here I think it is warranted.

Reply:
People do not always realize there is an inner exception.

Reply:
ok